### PR TITLE
Release v0.10.11

### PR DESCRIPTION
**Pull request by angrybayblade**
## Release summary

Version number: v0.10.11

## Release details

Autonomy:
- Adds support for minting services on layer 2 chains
- Adds support for skipping dependency checks when minting components

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] I've updated the dependencies versions to the latest, wherever is possible.
- [ ] Lint and unit tests pass locally (please run tests also manually, not only with `tox`)
- [ ] I built the documentation and updated it with the latest changes
- [ ] I've added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `__init__.py` file.
- [ ] I published the latest version on TestPyPI and checked that the following command work:
       ```pip install project-name==<version-number> --index-url https://test.pypi.org/simple --force --no-cache-dir --no-deps```
- [ ] After merging the PR, I'll publish the build also on PyPI. Then, I'll make sure the following
      command will work:
      ```pip install project-name-template==<version_number> --force --no-cache-dir --no-deps```  
- [ ] After merging the PR, I'll tag the repo with `v${VERSION_NUMVER}` (e.g. `v0.1.2`)

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: autonomy/chain/mint.py [hunk 1]
```diff
@@ -89,6 +91,7 @@ def wait_for_component_to_mint(
     deadline = datetime.datetime.now() + datetime.timedelta(seconds=timeout)
     while datetime.datetime.now() < deadline:
         token_id = token_retriever()
+        print(token_id)
```
Comment (DavidMinarsch) [inline autonomy/chain/mint.py:94]: remove

File: autonomy/chain/mint.py [hunk 2]
```diff
@@ -185,6 +188,7 @@ def mint_service(  # pylint: disable=too-many-arguments
     threshold: int,
     owner: Optional[str] = None,
     timeout: Optional[float] = None,
+    l2: bool = False,
```
Comment (DavidMinarsch) [inline autonomy/chain/mint.py:191]: Don't make this a flag. The rule is: Ethereum -> standard registry, all other networks -> l2 registry. No need for user having to know this.

File: autonomy/cli/helpers/chain.py [hunk 3]
```diff
@@ -159,6 +159,7 @@ def mint_component(  # pylint: disable=too-many-arguments, too-many-locals
     owner: Optional[str] = None,
     password: Optional[str] = None,
     skip_hash_check: bool = False,
+    skip_dependencies_check: bool = False,
```
Comment (DavidMinarsch) [inline autonomy/cli/helpers/chain.py:162]: This should also be set by the framework so user doesn't have to. Rule: if l2 (network other than ethereum) don't need it ever. If Network ethereum, give user choice.

File: docs/api/cli/mint.md [hunk 4]
```diff
@@ -194,6 +199,11 @@ Mint an agent.
     required=True,
     help="Threshold for the minimum numbers required to run the service",
 )
+@click.option(
```
Comment (DavidMinarsch) [inline docs/api/cli/mint.md:202]: See above; should be inferred from network selection. Not something user needs to bother with

**Review by DavidMinarsch — COMMENTED** [review]
Need a discussion around the two new packages. I think it would be best not to have new packages and instead discriminate inside each package.

File: autonomy/cli/helpers/chain.py [hunk 5]
```diff
@@ -199,20 +200,21 @@ def mint_component(  # pylint: disable=too-many-arguments, too-many-locals
...
-            ).contracts[chain_type],
-            dependencies=dependencies,
-            package_configuration=package_configuration,
-            skip_hash_check=skip_hash_check,
-        )
-    except FailedToRetrieveComponentMetadata as e:
-        raise click.ClickException(f"Dependency verification failed; {e}") from e
-    except DependencyError as e:
-        raise click.ClickException(f"Dependency verification failed; {e}") from e
+    if not skip_dependencies_check and chain_type == ChainType.ETHEREUM:
```
Comment (DavidMinarsch) [inline autonomy/cli/helpers/chain.py:203]: Do we want to also do 
```suggestion
    if not skip_dependencies_check and (chain_type == ChainType.ETHEREUM or chain_type == ChainType.GOERLI):
```

File: autonomy/cli/helpers/chain.py [hunk 6]
```diff
@@ -302,20 +305,21 @@ def mint_service(  # pylint: disable=too-many-arguments, too-many-locals
...
-            ).contracts[chain_type],
-            agent_id=agent_id,
-            service_configuration=cast(Service, package_configuration),
-            skip_hash_check=skip_hash_check,
-        )
-    except FailedToRetrieveComponentMetadata as e:
-        raise click.ClickException(f"Dependency verification failed; {e}") from e
-    except DependencyError as e:
-        raise click.ClickException(f"Dependency verification failed; {e}") from e
+    if not skip_dependencies_check and chain_type == ChainType.ETHEREUM:
```
Comment (DavidMinarsch) [inline autonomy/cli/helpers/chain.py:308]: see above

**Review by Adamantios — APPROVED** [review]

**Pull request merged**